### PR TITLE
Move x86 linux and osx sdist test to action.

### PR DESCRIPTION
**Pull request by trivialfis**
* Move sdist tests to github action
* Add windows test.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -59,6 +59,45 @@ jobs:
...
+      run: |
+        conda info
+        conda list
+    - name: Build and install XGBoost
+      shell: bash -l {0}
+      run: |
+        cd python-package
+        python --version
+        python setup.py sdist
+        pip install ./dist/xgboost-*.tar.gz
```
Comment (hcho3) [inline .github/workflows/main.yml:97]:
```suggestion
        pip install -v ./dist/xgboost-*.tar.gz
```
Add `-v` so that we can see all build steps.

**Review by hcho3 — CHANGES_REQUESTED** [review]
LGTM. One minor comment:

File: .github/workflows/main.yml (hunk 1 above)
Comment (trivialfis) [reply .github/workflows/main.yml]: Done.

**Comment by trivialfis** [comment]
@hcho3 For unknown reason the gcc on travis is downgraded from [7.5.0](https://travis-ci.org/github/dmlc/xgboost/jobs/755501923) to [7.3.1](https://travis-ci.org/github/dmlc/xgboost/jobs/756068981), now compilation is failing.  Any suggestion on how to workaround the failing arm test?  We used to fix the gcc version on travis, not sure if we should restore it now.

**Comment by hcho3** [comment]
I wonder if it's OOM error.

**Comment by hcho3** [comment]
We could disable the sdist test on ARM for now.

**Comment by trivialfis** [comment]
Python test is suffering from a different kind of error, it's not deterministic:

```
-- Build files have been written to: /workspace/build

[1/1] Cleaning all built files...

Cleaning... 0 files.

[1/163] /opt/rh/devtoolset-7/root/usr/bin/c++ -DDMLC_CORE_USE_CMAKE -DDMLC_UNIT_TESTS_USE_CMAKE -DDMLC_USE_CXX11=1 -D_DARWIN_C_SOURCE -D_POSIX_C_SOURCE=200809L -D_POSIX_SOURCE -D_XOPEN_SOURCE=700 -D__USE_XOPEN2K8 -I../dmlc-core/test/unittest -Igoogletest-src/googlemock/include -I../dmlc-core/include -Idmlc-core/include -isystem googletest-src/googletest/include -isystem googletest-src/googletest -O3 -DNDEBUG -fPIE -fopenmp -std=c++11 -MD -MT dmlc-core/test/unittest/CMakeFiles/dmlc_unit_tests.dir/unittest_main.cc.o -MF dmlc-core/test/unittest/CMakeFiles/dmlc_unit_tests.dir/unittest_main.cc.o.d -o dmlc-core/test/unittest/CMakeFiles/dmlc_unit_tests.dir/unittest_main.cc.o -c ../dmlc-core/test/unittest/unittest_main.cc

time="2021-01-25T11:24:37Z" level=error msg="error waiting for container: unexpected EOF"
```

**Comment by hcho3** [comment]
This one is using a Docker container, so the GCC version there is deterministic.

**Comment by trivialfis** [comment]
@hcho3 All passed.  Travis is back ?

**Pull request merged**